File: array/184.py
from typing import List


def specialArray(nums: List[int]) -> int:
    cant = len(nums)
    nums.sort()

    for index in range(len(nums)):
        if nums[index] == 0:
            cant -= 1
            continue

        if nums[index] >= cant and index - 1 > 0 and nums[index - 1] < cant:
            break

        if nums[index] >= cant and index - 1 < 0:
            break

        cant -= 1

    if cant != 0:
        return cant
    else:
        return -1


print(specialArray([0, 4, 3, 0, 4]))

File: array/184.cpp
#include <bits/stdc++.h>

using namespace std;

int specialArray(vector<int>& nums) {
    sort(nums.begin(), nums.end());
    int  n = nums.size();
    int cant = n;

    for(int i = 0; i < n; ++i) {
        if(nums[i] == 0) {
            --cant;
            continue;
        }

        if((nums[i] >= cant && i - 1 > 0 && nums[i - 1] < cant) || (nums[i] >= cant && i - 1 < 0))
            break;

        --cant;
    }

    return cant != 0 ? cant: -1;
}

int main() {
    vector<int> nums{0, 0, 1,2,3,4,5,6,7,8};
    cout<<specialArray(nums);

    return 0;
}